pkg/controller/workload/kubernetes/target/target_test.go

```diff
@@ -232,165 +191,18 @@ func TestReconcile(t *testing.T) {
-										Controller: &controller,
-									},
-								},
-							),
-								withConnectionSecretRef(&runtimev1alpha1.LocalSecretReference{Name: "super-secret"}),
-								withTargetLabels(map[string]string{"not": "dev"}))
-						}
-						return nil
-					},
-					MockUpdate: test.NewMockUpdateFn(nil, func(got runtime.Object) error {
-						want := kubeTarget(withOwnerReferences(
-							[]metav1.OwnerReference{
-								{
-									UID:        uid,
-									Name:       name,
-									Kind:       computev1alpha1.KubernetesClusterKind,
-									APIVersion: computev1alpha1.SchemeGroupVersion.String(),
-									Controller: &controller,
-								},
-							},
-						),
-							withConnectionSecretRef(&runtimev1alpha1.LocalSecretReference{Name: "super-secret"}),
-							withTargetLabels(map[string]string{"dev": "true"}))
-						if diff := cmp.Diff(want, got); diff != "" {
-							t.Errorf("-want, +got:\n%s", diff)
-						}
-						return nil
-					}),
-				},
-				log: logging.NewNopLogger(),
-			},
-			req:        reconcile.Request{NamespacedName: types.NamespacedName{Namespace: namespace, Name: name}},
-			wantResult: reconcile.Result{},
-			wantErr:    nil,
-		},
-		"SuccessfulUpdateExistingEmptyTarget": {
-			rec: &Reconciler{
-				kube: &test.MockClient{
-					MockGet: func(_ context.Context, key client.ObjectKey, obj runtime.Object) error {
-						switch key {
-						case client.ObjectKey{Namespace: namespace, Name: name}:
-							*obj.(*computev1alpha1.KubernetesCluster) = *kubeCluster(withWriteConnectionSecretToRef(
-								&runtimev1alpha1.LocalSecretReference{Name: "super-secret"},
-							), withLabels(map[string]string{"dev": "true"}))
-						case client.ObjectKey{Namespace: namespace, Name: targetName}:
-							*obj.(*workloadv1alpha1.KubernetesTarget) = *kubeTarget(withOwnerReferences(
-								[]metav1.OwnerReference{
-									{
-										UID:        uid,
-										Name:       name,
-										Kind:       computev1alpha1.KubernetesClusterKind,
-										APIVersion: computev1alpha1.SchemeGroupVersion.String(),
-										Controller: &controller,
-									},
-								},
-							))
-						}
-						return nil
+						}),
 					},
-					MockUpdate: test.NewMockUpdateFn(nil, func(got runtime.Object) error {
-						want := kubeTarget(withOwnerReferences(
-							[]metav1.OwnerReference{
-								{
-									UID:        uid,
-									Name:       name,
-									Kind:       computev1alpha1.KubernetesClusterKind,
-									APIVersion: computev1alpha1.SchemeGroupVersion.String(),
-									Controller: &controller,
-								},
-							},
-						),
-							withConnectionSecretRef(&runtimev1alpha1.LocalSecretReference{Name: "super-secret"}),
-							withTargetLabels(map[string]string{"dev": "true"}))
-						if diff := cmp.Diff(want, got); diff != "" {
-							t.Errorf("-want, +got:\n%s", diff)
-						}
+					Applicator: resource.ApplyFn(func(_ context.Context, _ runtime.Object, _ ...resource.ApplyOption) error {
 						return nil
```

Comment:
We could check to make sure that the `KubernetesTarget` actually points to the right secret and has the correct labels here. Otherwise, there isn't much of a point in setting the labels, etc. that we want propagated on the `KubernetesCluster` we return from the `MockGet`. Happy to defer this to a follow-up though 👍

Reply:
I should be able to address this now.